packages/clima_ui/lib/screens/weather_screen.dart

```diff
@@ -143,46 +175,63 @@ class WeatherScreen extends HookWidget {
                   }
                 }(),
               ),
-              child: Container(
-                constraints: const BoxConstraints.expand(),
-                child: SingleChildScrollView(
-                  physics: const BouncingScrollPhysics(),
-                  child: Column(
-                    children: [
-                      const MainInfoWidget(),
-                      Divider(
-                        color: Theme.of(context)
-                            .textTheme
-                            .subtitle1!
-                            .color!
-                            .withAlpha(65),
-                      ),
-                      SizedBox(
-                        height: MediaQuery.of(context).size.aspectRatio == 1.0
-                            ? 24.h
-                            : 16.h,
-                        child: const HourlyForecastsWidget(),
-                      ),
-                      Divider(
-                        color: Theme.of(context)
-                            .textTheme
-                            .subtitle1!
-                            .color!
-                            .withAlpha(65),
-                      ),
-                      const DailyForecastsWidget(),
-                      Divider(
-                        color: Theme.of(context)
-                            .textTheme
-                            .subtitle1!
-                            .color!
-                            .withAlpha(65),
-                      ),
-                      const AdditionalInfoWidget(),
-                    ],
-                  ),
-                ),
-              ),
+              child: fullWeatherState is w.Error &&
+                      fullWeatherState.fullWeather == null
+                  ? FailureWidget(
+                      failure: fullWeatherState.failure,
+                      onRetry: () async {
+                        await fullWeatherStateNotifier.loadFullWeather();
+                      },
```

Comment:
Same as above, please test first:
```suggestion
                      onRetry: fullWeatherStateNotifier.loadFullWeather,
```